src/tui.rs

```diff
@@ -0,0 +1,130 @@
+use crate::app::App;
+use crate::cloud_client::CloudClient;
+use crossterm::event;
+use crossterm::event::{Event, KeyCode, KeyEventKind};
+use ratatui::{
+    prelude::*,
+    widgets::{Block, Borders, List, Paragraph},
+};
+use std::io;
+
+pub enum WorkMode {
+    Read,
+    Edit,
+}
+
+pub fn ui<C: CloudClient>(frame: &mut Frame, app: &App<C>) {
+    let main_layout = Layout::vertical([
+        Constraint::Length(1),
+        Constraint::Length(3),
+        Constraint::Percentage(50),
+        Constraint::Percentage(50),
+    ]);
+    let [help_area, input_area, log_area, storages_area] = main_layout.areas(frame.size());
+
+    let (msg, style) = match app.work_mode {
+        WorkMode::Read => (
+            vec![
+                "Press ".into(),
+                "q".bold(),
+                " to exit, ".into(),
+                "e".bold(),
+                " to start editing.".bold(),
+            ],
+            Style::default().add_modifier(Modifier::RAPID_BLINK),
+        ),
+        WorkMode::Edit => (
+            vec![
+                "Press ".into(),
+                "Esc".bold(),
+                " to stop editing, ".into(),
+                "Enter".bold(),
+                " to record the message".into(),
+            ],
+            Style::default(),
+        ),
+    };
+    let text = Text::from(Line::from(msg)).patch_style(style);
+    let help_message = Paragraph::new(text);
+    frame.render_widget(help_message, help_area);
+
+    let input = Paragraph::new(app.input_command.as_str())
+        .style(match app.work_mode {
+            WorkMode::Read => Style::default().fg(Color::LightYellow),
+            WorkMode::Edit => Style::default().fg(Color::LightGreen),
+        })
+        .block(
+            Block::default()
+                .borders(Borders::ALL)
+                .title("Input command"),
+        );
+    frame.render_widget(input, input_area);
+    match app.work_mode {
+        WorkMode::Read => {}
+        WorkMode::Edit => {
+            #[allow(clippy::cast_possible_truncation)]
+            frame.set_cursor(
+                input_area.x + app.cursor_position as u16 + 1,
```

Comment:
I would better use explicit type conversion to make this failable, or change types so conversion would not be necessary.
```suggestion
                input_area.x + u16::try_from(app.cursor_position).expect("...") + 1,
```